.travis/travis-compile.sh

```diff
@@ -20,6 +20,29 @@ if [[ $BUILDTYPE == "Debug" ]]; then
 	cmake .. -DWITH_SERVER=1 -DTEST=1 -DCMAKE_BUILD_TYPE=$BUILDTYPE $prefix
 	make -j2
 	make test
+
+	clang-format -i \
+		common/*.h \
+		common/*.cpp \
+		cockatrice/src/*.h \
+		cockatrice/src/*.cpp \
+		oracle/src/*.h \
+		oracle/src/*.cpp \
+		servatrice/src/*.h \
+		servatrice/src/*.cpp
+	
+	git clean -f
+	git diff --quiet || (
+		echo "***************************************************";
+		echo "*** The code is not clean against clang-format  ***";
+		echo "*** Please run clang-format and fix the         ***";
+		echo "*** differences then rebase/squash them into    ***";
+		echo "*** the relevant commits. Do not add a commit   ***";
+		echo "*** for just formatting fixes. Thanks!          ***";
```

Comment:
Do we need to ask for that?

It makes it a bit more difficult for new contributors to have to deal with squashing/rebasing if they are not yet experienced with it.

On merging a PR we can simply squash commits over the GitHub PR interface on the webpage right before merging.

Reply:
> we can simply squash commits

Reply:
I think having travis fail and display a message/warning to tell the user what's up is good. 👍 

I would just adjust the message to something like:
```
echo "*****************************************************";
echo "***  This PR is not clean against our code style  ***";
echo "***  Run clang-format and fix up any differences  ***";
echo "***  Check our CONTRIBUTING.md file for details!  ***";
echo "***                  Thank you ♥                  ***";
echo "*****************************************************";
```
CONTRIBUTING.md should have some details on how to run and correct the source code, too.

(the heart might be difficult to print in bash, though... 😃 )